### ISSUE TITLE
stage_ros malloc(): memory corruption

### ISSUE THREAD

**Issue by nascimentocrafael** [ros-simulation/stage_ros#21]
Hi,

I'm trying to run `stage_ros` with the example worlds but I'm getting `malloc(): memory corruption` error.
I'm running on ubuntu 14.04 with ros indigo.

When I do: `rosrun stage_ros stageros -g  willow-erratic.world` It works fine.

But when I do: `rosrun stage_ros stageros willow-erratic.world` It gives me an error:

```
malloc(): memory corruption
```

Any ideas of the problem?

Thank you very much

**Comment by wjwwood** [comment]
It looks like the argument parsing is got a bug in it. For now I'd suggest just using `-g` before the world file. I don't have time to look into it at the moment, but I'd be happy to look at a pull request if someone else has time to reproduce and debug the issue.

**Comment by wjwwood** [comment]
For reference, it may have been introduced in this set of changes (that's the most recent change that touched the argument parsing):

https://github.com/ros-simulation/stage_ros/pull/10/files#diff-85d0e0c7dc1f3bd834819d2137cba2c0R717

**Comment by gerkey** [comment]
I'm unable to reproduce the problem. I tried the following configurations on my Ubuntu 14.04.3 machine:

* indigo from debs
* jade from debs
* `stage_ros` from source (master) on top of indigo debs

In all cases, `stageros` starts up correctly when passed `willow-erratic.world`, with `-g` and without `-g`. I don't see any crashes.

I also ran the source build of `stage_ros` under `valgrind` with and without `-g` and saw no errors (other than some uninitialized values coming from GL in the GUI case).

@nascimentocrafael, can you offer a more detailed method of reproduction?

**Comment by daenny** [comment]
The problem is very eluding, I have stage installed on three different machines and none produces that bug. A machine of a friend does produce it though, using the same Ubuntu as I do, and hydro installed from deb. I will see if I can find an installation which also produces this.

**Comment by gerkey** [comment]
@daenny  Please do let me know if you can describe how to reproduce it

**Comment by daenny** [comment]
Ok, so I found a machine which had that bug. It was an iMac 2013, 27inch. Running ubuntu natively:
`uname -a: 3.16.0-46-generic #62~14.04.1-Ubuntu SMP Tue Aug 11 16:27:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`
Nvidia drivers are from: ppa xorg-edgers
I did an update of all binaries (running indigo): `sudo apt-get update && sudo apt-get dist-upgrade` and pulled the newest master of stage_ros. Recompiled and the error is gone. I had to manually install ros-indigo-stage though for some reason, which also installed fltk1.1 alongside. So maybe that was not correctly install before. 
So to sum up: the bug was definitely there on this machine, i.e. starting the gui crashed with a memory corruption, but after a complete update, installing ros-indigo-stage (with libfltk1.1) and recompile it is gone on this machine as well... I will ask my friend to do the same on his machine(s) and test if its gone then as well.
I could imagine that the fltk dependency was somehow messed up before.

**Comment by gerkey** [comment]
Thanks for the follow-up. I'll close this ticket on the assumption that the problem (whatever the original cause) has disappeared. Re-open if you find otherwise and we'll keep investigating.
